pkg/process/monitor/process_monitor.go

```diff
@@ -114,114 +148,40 @@ func (p *ProcessMonitor) evalEXECCallback(c *ProcessCallback, pid uint32) {
 		if c.Regex.MatchString(pname) {
 			p.enqueueCallback(c, pid)
 		}
-	case MAPFILE:
-		mmaps, err := proc.MemoryMaps(true)
-		if err != nil {
-			log.Errorf("process %d maps parsing failed %s", pid, err)
-			return
-		}
-		if mmaps == nil {
-			return
-		}
-		for _, mmap := range *mmaps {
-			if c.Regex.MatchString(mmap.Path) {
-				p.enqueueCallback(c, pid)
-				break
-			}
-		}
 	}
 }
 
-func (pm *ProcessMonitor) Subscribe(callback *ProcessCallback) (UnSubscribe func(), err error) {
+// Initialize will scan all running processes and execute matching callbacks
+// Once it's done all new events from netlink socket will be processed by the main async loop
+func (pm *ProcessMonitor) Initialize() error {
 	pm.m.Lock()
 	defer pm.m.Unlock()
 
-	for _, c := range pm.procEventCallbacks[callback.Event] {
-		if c == callback {
-			return func() {}, errors.New("same callback can't be registred twice")
-		}
+	if pm.isInitialized {
+		return fmt.Errorf("Process monitor already initialized")
```

Comment:
maybe warning?

Reply:
It's a return error ...  `Initialize()` must only be call once during the life time (isInitialized == true)